File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit=0
        min_price=prices[0]
        for i in prices:
            current_price=i
            potential_profit=current_price-min_price
            max_profit=max(max_profit,potential_profit)
            min_price=min(min_price,current_price)
        return max_profit
        

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int minpr=prices[0];
        int pro=0;
        for(int i=1;i<prices.length;i++){
            if(prices[i]<minpr){
                minpr=prices[i];
            }
            int profit=prices[i]-minpr;
            if(profit>pro){
                pro=profit;
            }
        }
        return pro;
    }
}